modin/pandas/series.py

```diff
@@ -1003,6 +1009,15 @@ def factorize(self, sort=False, use_na_sentinel=True):  # noqa: PR01, RT01, D200
             use_na_sentinel=use_na_sentinel,
         )
 
+    def case_when(self, caselist):  # noqa: PR01, RT01, D200
```

Comment:
Can we put this in the query compiler and default to pandas there?

Reply:
We have a lot of places where we default to pandas at the API layer. I think it would be better to move the logic of d2p to the query compiler layer in a separate issue.

Reply:
In general, I don’t mind either way. Tell me if I need to separate the latest changes.